### ISSUE TITLE
Training loop

### ISSUE THREAD

**Issue by RussellALA** [stdhd/inverse-sketcher#3]
Implement a training loop with the necessary functionality:

- [x]  Optimizer
- [x]  Learning rate scheduling, weight decay
- [x]  Model saving and loading (Include Optimizer params and train/test split in saved state!)
- [x]  Training progress monitoring (TQDM)
- [x]  Rewrite scheduler to exponential decay
